src/components/menu/menu.tsx

```diff
@@ -150,28 +179,181 @@ export class Menu {
-                            class={{
-                                'has-grid-layout has-interactive-items':
-                                    this.gridLayout,
-                            }}
-                            items={this.items}
-                            type="menu"
-                            badgeIcons={this.badgeIcons}
-                            onSelect={this.handleSelect}
-                            ref={this.setListElement}
-                        />
+                        {this.renderBreadcrumb()}
+                        {this.renderMenuList()}
                     </limel-menu-surface>
                 </limel-portal>
             </div>
         );
     }
 
-    public componentDidRender() {
-        const slotElement = this.host.shadowRoot.querySelector('slot');
-        slotElement.assignedElements().forEach(this.setTriggerAttributes);
+    @Watch('items')
+    protected itemsWatcher() {
+        this.currentItems = null;
+
+        this.setFocus();
+    }
+
+    @Watch('currentItems')
+    protected currentItemsWatcher() {
+        this.setFocus();
+    }
+
+    @Watch('open')
+    protected openWatcher() {
+        if (!this.open) {
+            this.currentItems = null;
+
+            return;
+        }
+
+        this.setFocus();
     }
 
+    private renderBreadcrumb = () => {
+        if (!this.currentSubMenu) {
+            return;
+        }
+
+        const breadCrumbItems: MenuCrumbItem[] = [];
+        let currentItem = this.currentSubMenu;
+        while (currentItem) {
+            breadCrumbItems.unshift({
+                text: currentItem.text,
+                icon: {
+                    name: currentItem.icon,
+                    color: currentItem.iconColor,
+                },
+                menuItem: currentItem,
+            });
+            currentItem = currentItem.parentItem;
+        }
+
+        if (!breadCrumbItems.length) {
+            return;
+        }
+
+        return (
+            <limel-breadcrumbs
+                style={{
+                    'border-bottom': 'solid 1px rgb(var(--contrast-500))',
+                    'flex-shrink': '0',
+                }}
+                onSelect={(
+                    event: LimelBreadcrumbsCustomEvent<MenuCrumbItem>
+                ) => {
+                    if (!event.detail.menuItem) {
+                        this.currentItems = null;
+                        this.currentSubMenu = null;
+                        this.navigateMenu.emit(null);
+
+                        return;
+                    }
+
+                    this.handleSelect(event.detail.menuItem);
+                }}
+                items={[
+                    {
+                        text: '',
+                        icon: {
+                            name: 'home',
+                        },
+                        type: 'icon-only',
+                    },
+                    ...breadCrumbItems,
+                ]}
```

Comment:
If we optimise the breadcrumbs-creating loop so it doesn't have to be run on every single render, we should move this out too, for the same reason.

We should be able to update the breadcrumbs when navigation happens, and just keep that state between renders.

Reply:
Yeah, before the logic for this was very different, but now when we have the `currentSubMenu` in a prop it's easier to do that. I can try it out.

Reply:
⚡

Reply:
Or, well, no, not quite zapped. I've just improved it a little bit. It still runs on every render.

Reply:
Sorry, I'm too tired and don't read properly. This is indeed zapped, because, while the loop is still run on every render, this specific thing has been moved, so it's done next to the loop. So I'll resolve this after all. Sorry for spamming… 🙇‍♂️